code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

```diff
@@ -0,0 +1,200 @@
+#define MOB_SIZE_LARGE 3
+#define LARGE_MOB_LAYER 4.4
+
+/mob/living/simple_animal/hostile/megafauna
+	name = "boss of this gym"
+	desc = "Attack the weak point for massive damage."
+	health = 1000
+	maxHealth = 1000
+	a_intent = I_HURT
+	environment_smash = ENVIRONMENT_SMASH_WALLS
+	light_range = 3
+	faction = list("mining", "boss")
+	var/robust_searching = TRUE //By default, mobs have a simple searching method, set this to 1 for the more scrutinous searching (stat_attack, stat_exclusive, etc), should be disabled on most mobs
+	var/stat_attack = DEAD
+	var/atom/target
+	//atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
```

Comment:
There's a lot of commented out stuff in here too that should be cleaned up before this gets merged.